esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -431,7 +432,8 @@ void EsdlServiceImpl::configureUrlMethod(const char *method, IPropertyTree &entr
     entry.setProp("@prot", protocol);
     entry.setProp("@path", path);
 
-    Owned<ISmartSocketFactory> sf = createSmartSocketFactory(iplist, true);
+    Owned<ISmartSocketFactory> sf = new CSmartSocketFactory(iplist, true);
```

Comment:
To ensure we're instantiating a SmartSocket, we need to use the smartsocket smartendpoint to use access the hostname (if one exists) the ISmarSocketFactory interface is not aware of the smartendpoints. We really didn't need to use a factory pattern for these connections to begin with.